board/tsat/3500/common/rootfs_overlay/usr/local/bin/load_terminal.sh

```diff
@@ -11,12 +11,30 @@ NEXT_BASE_DIR="$BASE_DIR/next"
 CURRENT_RUN_SCRIPT="$CURRENT_BASE_DIR/$RUN_SCRIPT"
 NEXT_RUN_SCRIPT="$NEXT_BASE_DIR/$RUN_SCRIPT"
 
+MAX_TRIES=3
+REMAINING_TRIES_FILE="/tmp/term_start_tries"
+
 # do nothing if noauto flag is present
 if [ -f "$NOAUTO_FLAG" ]; then
   sleep 30
   exit 0
 fi
 
+# give up if term crashes repeatedly
+if [ -f "$REMAINING_TRIES_FILE" ]; then
+  REMAINING_TRIES=($(<"$REMAINING_TRIES_FILE"))
```

Comment:
fancy, how does it work?

Reply:
Uses bash's built-in file redirection. The input file will be broken up according to your IFS settings (space/tab/newline by default). Each "word" in the input file will be stored as one entry in the array.

Reply:
is it Bash-specific or does it work with 'sh' as well? (since we don't have Bash on target)

Reply:
if it doesn't work on target you could use something like:

```sh
read REMAINING_TRIES < "$REMAINING_TRIES_FILE"
```